src/LintingViewExtension/Controls/GraphRuleIssueDto.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections.ObjectModel;
+using Dynamo.Linting.Rules;
+
+namespace Dynamo.LintingViewExtension.Controls
+{
+    public class GraphRuleIssueDto
```

Comment:
if this really needs to be public can you add a summary, does this really need to be public ? and can you add a summary anyway because I don't know if everyone will be familiar with a dto. Usually I am familiar with them in context of a server - is there an analogous design here?

Reply:
This does not need to be public no.

In terms of the name, I think I just hit the big old "Don't know what to call this" wall!
I'll revisit it and try to make the name more obvious.

Reply:
Might have been a bit too quick with the not needing to be public thing..

As I'm binding a collection of these issues in the view that property needs to be public and therefore it can't be a collection of internal types. I made all Rule issues share a common interface `IRuleIssue` and make the public collections of type `IRuleIssue`.

This means that both `GraphRuleIssue` and `NodeRuleIssue` can be internal. The `IRuleIssue` will still need to be public though.